components/function-controller/internal/webhook/validating_webhook.go

```diff
@@ -16,17 +16,19 @@ import (
 )
 
 type ValidatingWebHook struct {
-	configv1alpha1 *serverlessv1alpha1.ValidationConfig
-	configv1alpha2 *serverlessv1alpha2.ValidationConfig
-	client         ctrlclient.Client
-	decoder        *admission.Decoder
+	configv1alpha1       *serverlessv1alpha1.ValidationConfig
+	configv1alpha2       *serverlessv1alpha2.ValidationConfig
+	client               ctrlclient.Client
+	decoder              *admission.Decoder
+	rejectJava17AlphaJVM bool
 }
 
-func NewValidatingHook(configv1alpha1 *serverlessv1alpha1.ValidationConfig, configv1alpha2 *serverlessv1alpha2.ValidationConfig, client ctrlclient.Client) *ValidatingWebHook {
+func NewValidatingWebHook(configv1alpha1 *serverlessv1alpha1.ValidationConfig, configv1alpha2 *serverlessv1alpha2.ValidationConfig, featureFlags FeatureFlags, client ctrlclient.Client) *ValidatingWebHook {
 	return &ValidatingWebHook{
-		configv1alpha1: configv1alpha1,
-		configv1alpha2: configv1alpha2,
-		client:         client,
+		configv1alpha1:       configv1alpha1,
+		configv1alpha2:       configv1alpha2,
+		client:               client,
+		rejectJava17AlphaJVM: !featureFlags.Java17JvmAlphaEnabled,
```

Comment:
Why are we using a negated field, and not the feature flag value directly?

Reply:
In context of webhook, the runtime can be rejected. 
I've done it for readability.